### PR TITLE
Popup text entry works on the index and step screens

### PR DESCRIPTION
**Pull request by mmcbee1**
Text box appears when the 'input results' button is clicked on the test run index and individual step screens. The size of the popup scales based on the device's screen size. When the keyboard appears, the popup moves upwards to accommodate. Multi-line text can be entered, which saves when the 'done' button is pressed. This area will scroll when more text is entered than will fit the view. The keyboard's return key enters a newline. Entered text on the index screen will persist until the run is finished/saved or cancelled, even when moving between multiple step screens.

Things that may be issues:
* While testing in the simulator, sometimes the keyboard will not appear on the first click of the text view and you need to hit command + K. If this is not done, popup placement will be wrong after 'done' button is entered. Not sure if this is a bug or just a simulator issue.
* Right now the same functionality is implemented in both the TestRunIndexViewController and the TestStepViewController as well as the storyboards for both. I had originally planned to create a new storyboard view and associated view controller but had a few issues and ran out of time. Since this functionality is blocking our progress and I would rather add what I've finished now, but would love some assistance refactoring.
* Currently no unit tests/UI tests

### CODE REVIEW

File: Mobile Test Runner for Jama/Mobile Test Runner for Jama/TestStepViewController.swift [hunk 1]
```diff
@@ -47,4 +54,43 @@ class TestStepViewController: UIViewController {
...
+    // Move popup when keyboard appears/hides
+    func keyboardWillShow(notification: NSNotification) {
+        if let keyboardSize = (notification.userInfo?[UIKeyboardFrameBeginUserInfoKey] as? NSValue)?.cgRectValue {
+            self.inputResultsBox.frame.origin.y -= keyboardSize.height/3
+        }
+    }
+    
+    func keyboardWillHide(notification: NSNotification) {
+        if let keyboardSize = (notification.userInfo?[UIKeyboardFrameBeginUserInfoKey] as? NSValue)?.cgRectValue {
+            self.inputResultsBox.frame.origin.y += keyboardSize.height/3
```
Comment (dwagg) [inline Mobile Test Runner for Jama/Mobile Test Runner for Jama/TestStepViewController.swift:112]: In this method I think we need to set the step's result. Currently we are saving whatever is put into the input box into a string but that is never making its way back into the actual step model. I think we should add another function into the protocol defined in the TestRunIndexViewController, something like `didSetStepResult(result: String)` where it sets the steps result value. then call that function from in this method. `indexDelegate.didSetStepResult(result: self.stepResult)`

**Review by dwagg — CHANGES_REQUESTED** [review]
I have added a comment in the TestStepViewController to preserve the results value if the user selects a status or presses the back button.

**Comment by dwagg** [comment]
All the code seems to make sense. I don't think this is very testable in unit tests at least from what I can tell so it looks good to me. Once the change that I requested is done I think it is good to go.

**Review by dwagg — APPROVED** [review]

**Review by JasonRitz — APPROVED** [review]
Everything looks good and it works as expected. Good Job!!

**Pull request merged**